### PR TITLE
chore(Makefile): update go-dev to 0.10.0 and compress registry binary

### PR DESCRIPTION
**Pull request by mboersma**

### CODE REVIEW

**Comment by mboersma** [comment]
Updated to use `deis/go-dev:0.9.1`.

File: Makefile [hunk 1]
```diff
@@ -52,7 +53,9 @@ docker-push: check-docker check-registry
 	docker push ${IMAGE}
 
 build-binary:
-	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -a -installsuffix cgo -ldflags ${LDFLAGS} -o $(BINDIR)/${SHORT_NAME} main.go
+	${DEV_ENV_PREFIX} -e CGO_ENABLED=0 ${DEV_ENV_IMAGE} go build -a -installsuffix cgo -ldflags ${LDFLAGS} -o $(BINDIR)/${SHORT_NAME} main.go || exit 1
+	$(call check-static-binary,$(BINDIR)/${SHORT_NAME})
+	${DEV_ENV_PREFIX} ${DEV_ENV_IMAGE} goupx --strip-binary -9 --brute $(BINDIR)/${SHORT_NAME} || exit 1
```
Comment (krancour) [inline Makefile]: What's the reason for the `|| exit 1`?  At first glance, it appears unnecessary.

Comment (mboersma) [inline Makefile]: It does, yes... I copy-pasted this from elsewhere. I think the `|| exit 1` was originally to break out of a Makefile loop correctly?

I agree it looks crufty and I'll have a go at removing it.

**Comment by mboersma** [comment]
Updated to use [`deis/go-dev:0.10.0`](https://github.com/deis/docker-go-dev/releases/tag/0.10.0).

**Pull request merged**